src/status_im2/contexts/emoji_picker/data.cljs

```diff
@@ -0,0 +1,112 @@
+(ns status-im2.contexts.emoji-picker.data
+  (:require [status-im2.contexts.emoji-picker.constants :as constants]
+            [utils.transforms :as transforms]))
+
+;; The dataset is grouped by number to reduce the file size.
+;; It is then categorized below:
+;; - people (0 and 1)
+;; - nature (3)
+;; - food (4)
+;; - activity (6)
+;; - travel (5)
+;; - objects (7)
+;; - symbols (8)
+;; - flags (9)
+;;
+;; NOTE: smileys & emoticons (0) and people (1) are grouped in one section
+
+(def emoji-data (transforms/js->clj (js/require "../src/js/data/emojis.json")))
+
+(def list-data
+  [{:title :t/emoji-people
+    :icon  :i/faces ;; 0 and 1
+    :id    :people
+    :data  []}
+   {:title :t/emoji-nature
+    :icon  :i/nature ;; 3
+    :id    :nature
+    :data  []}
+   {:title :t/emoji-food
+    :icon  :i/food ;; 4
+    :id    :food
+    :data  []}
+   {:title :t/emoji-activity
+    :icon  :i/activity ;; 6
+    :id    :activity
+    :data  []}
+   {:title :t/emoji-travel
+    :icon  :i/travel ;; 5
+    :id    :travel
+    :data  []}
+   {:title :t/emoji-objects
+    :icon  :i/objects ;; 7
+    :id    :objects
+    :data  []}
+   {:title :t/emoji-symbols
+    :icon  :i/symbols ;; 8
+    :id    :symbols
+    :data  []}
+   {:title :t/emoji-flags
+    :icon  :i/flags ;; 9
+    :id    :flags
+    :data  []}])
+
+(def categorized-and-partitioned
+  (->> (reduce
+        (fn [acc {:keys [group] :as emoji}]
+          (case group
+
+            0
```

Comment:
perhaps we use some constants in this file to avoid the need for these indexs
e.g 
```
(def emoji-people-1 0)
(def emoji-people-2 1)
(def emoji-nature 2)
``` 
we can then remove the comments in the data struct above too 👍

Reply:
also then we could simplify this function a bit
```
(def emoji-people-1 0)
(def emoji-people-2 1)
(def emoji-nature 3)

(def groups [emoji-people-1
             emoji-people-2
             emoji-nature])

(def categorized-and-partitioned
  (->> (reduce
        (fn [acc {:keys [group] :as emoji}]
          (update-in acc [(nth groups group) :data] conj emoji))
        list-data
        emoji-data)
       (reduce (fn [acc {:keys [data] :as item}]
                 (conj acc (assoc item :data (partition-all constants/emojis-per-row data))))
               [])))
```
i.e doing the look up of the index from this groups collection 
`(nth groups group)`